cardinal/world.go

```diff
@@ -393,10 +393,23 @@ func (w *World) startServer() {
 	}()
 }
 
+//nolint:gocognit // will fix in the future
```

Comment:
The fix for this lint would be to move a chunk of self-contained logic into a helper function. 

I don't see this single-use function as cluttering the world object; it's a sensible isolation of some logic. 

What about renaming "emitResourceWarnings" to "emitManagerWarnings" or "logEmptyManagers"?

Reply:
the bulk of the complexity is from what comes after, so that's what should have been addressed.

splitting these simple self explanatory logs to a separate method makes code that you could've easily glanced over much harder to read.